CollapseLauncher/Classes/GameManagement/GamePlaytime/RegistryClass/CollapsePlaytime.cs

```diff
@@ -226,5 +249,153 @@ public void AddMinute()
+
+        public DateTime LastDbUpdate = DateTime.MinValue;
+        
+        // value store
+        private string? _jsonDataDb;
+        private double? _totalTimeDb;
+        private double? _lastPlayedDb;
+        private int?    _unixStampDb;
+        
+        // Key names
+        private string KeyPlaytimeJson => $"{_gameVersion.GameType.ToString()}-{_gameVersion.GameRegion}-pt-js";
+        private string KeyTotalTime    => $"{_gameVersion.GameType.ToString()}-{_gameVersion.GameRegion}-pt-total";
+        private string KeyLastPlayed   => $"{_gameVersion.GameType.ToString()}-{_gameVersion.GameRegion}-pt-lastPlayed";
+        private string KeyLastUpdated  => $"{_gameVersion.GameType.ToString()}-{_gameVersion.GameRegion}-pt-lu";
+        
+
+        #region Sync Methods
+        /// <summary>
+        /// Sync from/to DB at init
+        /// </summary>
+        /// <returns>true if require refresh, false if dont.</returns>
+        public async ValueTask<(bool IsUpdated, CollapsePlaytime? PlaytimeData)> DbSync()
+        {
+            LogWriteLine("[CollapsePlaytime::DbSync] Starting sync operation...", LogType.Default, true);
+            try
+            {
+                // Fetch database last update stamp
+                var stampDbStr = await DbHandler.QueryKey(KeyLastUpdated);
+                _unixStampDb     = !string.IsNullOrEmpty(stampDbStr) ? Convert.ToInt32(stampDbStr) : null;
+                
+                // Compare unix stamp from config
+                var unixStampLocal = Convert.ToInt32(DbConfig.GetConfig(KeyLastUpdated).ToString());
+                if (_unixStampDb == unixStampLocal) 
+                {
+                    LogWriteLine("[CollapsePlaytime::DbSync] Sync stamp equal, nothing needs to be done~", LogType.Default, true);
+                    return (false, null); // Do nothing if stamp is equal
+                }
+
+                // When Db stamp is newer, sync from Db
+                if (_unixStampDb > unixStampLocal)
+                {
+                    // Pull values from DB
+                    await UpdatePlaytime_Database_Pull();
+                    if (!_isDbPullSuccess) 
+                    {
+                        LogWriteLine("[CollapsePlaytime::DbSync] Database pull failed, skipping sync~", LogType.Error);
+                        return (false, null); // Return if pull failed
+                    }
+                    
+                    if (string.IsNullOrEmpty(_jsonDataDb))
+                    {
+                        LogWriteLine("[CollapsePlaytime::DbSync] _jsonDataDb is empty, skipping sync~", default, true);
+                        return (false, null);
+                    }
+                    LogWriteLine("[CollapsePlaytime::DbSync] Database data is newer! Pulling data~", LogType.Default, true);
+                    CollapsePlaytime? playtimeInner = _jsonDataDb.Deserialize(UniversalPlaytimeJSONContext.Default.CollapsePlaytime,
```

Comment:
Deserialization doesn't seem to be working, always returning an instance with default values.

Reply:
i can replicate it, but im not sure on why this is the case as the return bytes of the db query is perfect (in my eyes)
Example of json query result
```
7B 22 44 61 69 6C 79 50 6C 61 79 74 69 6D 65 22 3A 22 30 30 3A 34 30 3A 30 30 22 2C 22 57 65 65 6B 6C 79 50 6C 61 79 74 69 6D 65 22 3A 22 32 33 3A 30 30 3A 30 30 22 2C 22 4D 6F 6E 74 68 6C 79 50 6C 61 79 74 69 6D 65 22 3A 22 32 36 3A 30 30 3A 30 30 22 2C 22 4C 61 73 74 53 65 73 73 69 6F 6E 22 3A 22 30 30 3A 30 30 3A 30 30 22 2C 22 43 6F 6E 74 72 6F 6C 44 61 74 65 22 3A 22 32 30 32 34 2D 31 30 2D 31 33 54 30 30 3A 30 30 3A 30 30 2B 30 37 3A 30 30 22 7D
```
![image](https://github.com/user-attachments/assets/01363aa2-6520-47f1-822d-692bb5be2e10)


I'm really not sure how this happened now since it was fine when me and @neon-nyan tested it a few weeks ago (i think we tested it)

Reply:
this was actually my fault, it was already broken in #584
should be working now